app/tests/src/test_generate.py

```diff
@@ -89,8 +21,9 @@ def test_get_models_ollama(monkeypatch):
     if "OPENAI_API_KEY" in os.environ:
         monkeypatch.delenv("OPENAI_API_KEY")
     monkeypatch.setenv("OLLAMA_HOST", "mock_key")
-    monkeypatch.setattr(ollama, "list", ollama_model_list)
 
+    sys.modules["ollama"] = __import__("mock_ollama")
+    # TODO: use `monkeypatch_module` so that sys.modules doesn't mess up other tests
```

Comment:
@KevinJBoyer @ccheng26 Do y'all know how to use https://github.com/navapbc/labs-decision-support-tool/blob/5a46fd63f2788991f075e4a5088ffec4df148450/app/tests/conftest.py#L62

Reply:
I'm generally familiar, but did you see my comment [here](https://github.com/navapbc/labs-decision-support-tool/pull/23#issuecomment-2228899367)?

Reply:
If you're moving ollama to [tool.poetry.group.dev.dependencies], do you need to change anything about this test? It looks like moto is part of the same group, and is imported in tests

Reply:
> do you need to change anything about this test?

I don't think so. I just need to address the `TODO` I added.

Reply:
Played around with this -- if you're comfortable with `import ollama` in `test_generate.py`, this test will continue to work as originally written. However, you can remove the import by mocking the function using a string for the path, like: `monkeypatch.setattr("ollama.list", ollama_list)`.

Unfortunately, mocking the module is tricky because this PR modifies the code in `generate.py` to import the module within a function, so there's no way to do a top-level mock which you would normally do via `monkeypatch.setattr("src.generate.ollama", tests.mock.mock_ollama)`

Reply:
Reverted test files.